backend.native/tests/interop/objc/allocException.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+import platform.Foundation.*
+import platform.objc.*
+import kotlin.test.*
+
+import kotlinx.cinterop.*
+import platform.posix.*
+import kotlin.system.exitProcess
+
+fun exc_handler(x: Any?) : Unit {
+    println("Uncaught exception handler")
+    println(x.toString())
+    exitProcess(0)
+}
+
+fun main() {
+    objc_setUncaughtExceptionHandler(staticCFunction(::exc_handler))
+
+    // The following should fail with exception that shall be processed by FilterException and stop at exc_handler
+    // Otherwise, ReportUnhandledException will fail with Segmentation fault (exit code 139) on attempt to printStackTrace
```

Comment:
These implementation details are somewhat unrelated to the test itself.

Reply:
1. Arguable. This comment provides additional context for a test maintainer: the test relies on the fact that FilterException annotation will finish with std::terminate, and std::set_terminate hook is the default ObjC hook. This is not necessary be true: someone may decide to rework KonanTerminateHandler so it does not call oldTerminateHandler (see TerminateWithUnhandledException for example). It may exit with abort() (as TerminateWithUnhandledException does), exit(128) or any other way. Exception processing is still OK but hook has not been invoked and test failed. In that case a  __test__  itself shall be reworked, not the main logic. So the comment provides this kind of clue.
2. Can I ask you to provide all the comments at once, or leave a notice when you are not finished yet, so i won't bother to iterate (i.e. commit) until all feedbacks are collected and processed? The lines we're talking about were in the previous commit too.

Reply:
> This comment provides additional context for a test maintainer

Please note the huge difference between the behavior description, e.g.
> // Exception thrown by Objective-C `alloc` terminates the program and detected as uncaught with the hook above

and the description of the implementation:
> // The following should fail with exception that shall be processed by FilterException and stop at exc_handler
> // Otherwise, ReportUnhandledException will fail with Segmentation fault (exit code 139) on attempt to printStackTrace

I find the first one better serving the purpose you use as argument.


> The lines we're talking about were in the previous commit too.

This is definitely wrong: these lines have been significantly rewritten in the last commit by this moment: https://github.com/JetBrains/kotlin-native/pull/3318/commits/656a09026c562107010478e88e57244202961c99#diff-022d0c29e3a63ba37d377c862c10c450R23

> Can I ask you to provide all the comments at once, or leave a notice when you are not finished yet

I have to refuse, because
1. This is incompatible with proper review process: what if I notice something later, after I've reviewed the code for the first time? Do you suggest me to discard this comment?
2. There is no much sense in providing minor review comments (like this one) along with major ones (like using `objc_setUncaughtExceptionHandler`), since fixing the latter ones may (and did in this case) also affect the former ones.